app/service/planning_svc.py

```diff
@@ -7,16 +7,20 @@ class PlanningService(BasePlanningService):
     def __init__(self):
         self.log = self.add_service('planning_svc', self)
 
-    async def get_links(self, operation, phase=None, agent=None, trim=True):
+    async def get_links(self, operation, phase=None, agent=None, trim=True, stopping_conditions=[]):
         """
         For an operation, phase and agent combination, create links (that can be executed).
         When no agent is supplied, links for all agents are returned
         :param operation:
         :param phase:
         :param agent:
+        :param stopping_conditions:
         :param trim: call trim_links() on list of links before returning
         :return: a list of links
         """
+        if await self.check_stopping_conditions(operation, stopping_conditions):
```

Comment:
won't this get hit for each phase of the op?

Reply:
Yes?

Reply:
clarification ;) can't we hit it once per operation? if we reach the stopping condition on phase 2 of a 7 phase operation, might as well tie things off... 

I think we could do that fairly easily - by just incrementing the operation.phase property to the final phase (right here) and then in the main operation loop (in app_svc) changing the loop to check the phase #

Reply:
No you cannot hit it once per operation. When a phase ends and we've generated new facts, we need to check stopping conditions to see if we've found facts that meet the stopping condition.  Checking once per operation wouldn't make sense.  Incrementing the _phase_ to final phase once we've found the stopping condition makes sense though

Reply:
yeah-- that was my thought (I wasn't real clear). check every phase but once hit, move the phase to the end and close out.

Reply:
@khyberspache @privateducky I'm not so sure that we'll be able to increment the phase so easily as the phase iteration loop is implemented in app_svc. Am I missing something? 

Nevertheless, I pushed up a suggested implementation that accomplishes this. Let me know what your thoughts are.